couler/core/utils.py

```diff
@@ -274,3 +276,8 @@ def bool_to_str(bool_val):
     if not isinstance(bool_val, bool):
         raise TypeError("The bool_val is required to be boolean type")
     return "true" if bool_val else "false"
+
+
+def get_uniq_step_id():
+    global STEP_ID
```

Comment:
Why is this needed? Could we move this to states.py?

Reply:
Moved to `proto_repr.py` since it's only used in that file.